src/app/api/tasks/tasks.service.ts

```diff
@@ -0,0 +1,96 @@
+import { CreateTaskRequest, UpdateTaskRequest } from '@/types/dto/tasks.dto'
+import { BaseService } from '@api/core/services/base.service'
+import { Resource } from '@api/core/types/api'
+import { PoliciesService } from '@api/core/services/policies.service'
+import { AssigneeType } from '@prisma/client'
+
+type FilterByAssigneeId = {
+  assigneeId: string
+  assigneeType: AssigneeType
+}
+
+export class TasksService extends BaseService {
+  private buildReadFilters() {
+    const user = this.user
+
+    let filters = { where: { workspaceId: user.workspaceId, OR: undefined as unknown as FilterByAssigneeId[] } }
```

Comment:
what is this `undefined as unknown as FilterByAssigneeId[]`? trying to do?

Reply:
I wanted an `undefined` `OR` value for `where` so I could later inject it with `OR` condition if required. If `OR` condition is not required, the `undefined` OR value would cause no `OR`s to be generated in the final SQL query.

I admit it involves a bit of type gymnastics but i found it to be the shortest way to do this. I'm open to new solutions if required!

Reply:
I've added some method documentation here so we don't forget how this works in the future!

Reply:
and why do we need to typecast it to assigneeId[]?